packages/runtime-dom/src/components/Transition.ts

```diff
@@ -165,15 +165,15 @@ function normalizeDuration(
   if (duration == null) {
     return null
   } else if (isObject(duration)) {
-    return [toNumber(duration.enter), toNumber(duration.leave)]
+    return [NumberOf(duration.enter), NumberOf(duration.leave)]
   } else {
-    const n = toNumber(duration)
+    const n = NumberOf(duration)
     return [n, n]
   }
 }
 
-function toNumber(val: unknown): number {
-  const res = Number(val || 0)
+function NumberOf(val: unknown): number {
```

Comment:
When val = string => res = NaN => throw  is NaN.   This should throw isn't a number.
When val = NaN => res = 0 => check pass. This shold throw is NaN.